### PR TITLE
Add more output buffer stats and update in TaskStats

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Add bufferedBytes, bufferedPages, totalRowsSent and totalPagesSent in
OutputBuffer::Stats to be aligned with Java. We also add OutputBuffer::Stats
into TaskStats so Prestissimo can report this to coordinator as live output stats.

### CODE REVIEW

**Comment by mbasmanova** [comment]
CC: @HolyLow

File: velox/exec/OutputBuffer.h [hunk 1]
```diff
@@ -168,22 +168,34 @@ class OutputBuffer {
...
+    core::PartitionedOutputNode::Kind kind;
 
     /// States of the OutputBuffer.
-    const bool noMoreBuffers;
-    const bool noMoreData;
-    const bool finished;
+    bool noMoreBuffers{false};
+    bool noMoreData{false};
+    bool finished{false};
+    int64_t bufferedBytes{0};
```
Comment (mbasmanova) [inline velox/exec/OutputBuffer.h:192]: Are these sums of the buffersStats? Would you be nice to add a comment.

File: velox/exec/OutputBuffer.cpp [hunk 2]
```diff
@@ -358,11 +358,29 @@ void OutputBuffer::addOutputBuffersLocked(int numBuffers) {
...
+  bufferedBytes_ -= pageBytes;
+  VELOX_CHECK_GE(bufferedBytes_, 0);
+  bufferedPages_ -= numPages;
+  VELOX_CHECK_GE(bufferedPages_, 0);
+}
+
 bool OutputBuffer::enqueue(
     int destination,
-    std::unique_ptr<SerializedPage> data,
+    std::unique_ptr<SerializedPage> page,
```
Comment (mbasmanova) [inline velox/exec/OutputBuffer.cpp:381]: Meng, these refactorings are nice, but they may it very hard to review the actual changes. Would you split extract these into a separate PR?

File: velox/exec/TaskStats.h [hunk 3]
```diff
@@ -97,6 +98,9 @@ struct TaskStats {
   /// Indicates if output buffer is over-utilized and thus blocks the producers.
   bool outputBufferOverutilized{false};
 
+  /// Output buffer stats if not null.
```
Comment (mbasmanova) [inline velox/exec/TaskStats.h:101]: "not null" -> present

**Review by mbasmanova — COMMENTED** [review]
@xiaoxmeng Thank you for the change. Mixed in refactoring makes it very hard to review though. Any chance refactoring can be split into a separate PR?

File: velox/exec/OutputBuffer.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply velox/exec/OutputBuffer.cpp]: Reverted.

File: velox/exec/OutputBuffer.h (hunk 1 above)
Comment (xiaoxmeng) [reply velox/exec/OutputBuffer.h:194]: Yes, commented.

**Review by mbasmanova — APPROVED** [review]
Thanks.
